api/src/main/kotlin/com/gmtkgamejam/search/Opensearch.kt

```diff
@@ -0,0 +1,90 @@
+package com.gmtkgamejam.search
+
+import com.fasterxml.jackson.databind.node.ObjectNode
+import com.gmtkgamejam.models.posts.SearchItem
+import org.apache.hc.core5.http.HttpHost
+import org.opensearch.client.json.jackson.JacksonJsonpMapper
+import org.opensearch.client.opensearch.OpenSearchClient
+import org.opensearch.client.opensearch._types.SortOptions
+import org.opensearch.client.opensearch._types.query_dsl.Query
+import org.opensearch.client.opensearch.core.DeleteRequest
+import org.opensearch.client.opensearch.core.IndexRequest
+import org.opensearch.client.opensearch.core.SearchRequest
+import org.opensearch.client.opensearch.core.UpdateRequest
+import org.opensearch.client.transport.OpenSearchTransport
+import org.opensearch.client.transport.httpclient5.ApacheHttpClient5TransportBuilder
+
+/**
+ * Singleton access point to the Opensearch cluster
+ */
+object OpenSearch {
+
+    private val client: OpenSearchClient
+
+    init {
+        val transport: OpenSearchTransport = ApacheHttpClient5TransportBuilder
+            .builder(HttpHost("http", "localhost", 9200))
```

Comment:
Not sure if intended to hardcode it at localhost, if it is feel free to ignore this comment, if not, adding this into the .env could spare you some headaches when trying to deploy this.

Reply:
What a muppet am I - migrated to environment variables